src/interface/web/app/components/loginPrompt/loginPrompt.tsx

```diff
@@ -1,47 +1,536 @@
+        },
+    ];
+
+    return (
+        <div className="flex flex-col gap-4 p-4 md:p-0">
+            {!isMobileWidth && (
+                <Carousel
+                    plugins={[plugin.current]}
+                    className="w-full"
+                    onMouseEnter={() => {
+                        plugin.current.stop();
+                        setShowArrows(true);
+                    }}
+                    onMouseLeave={() => {
+                        plugin.current.play();
+                        setShowArrows(false);
+                    }}
+                >
+                    <CarouselContent>
+                        {tips.map((tip, index) => (
+                            <CarouselItem key={index}>
+                                <div className="relative p-0">
+                                    <Card>
+                                        <CardContent className="flex flex-col items-center justify-center rounded-b-none rounded-t-lg p-0">
+                                            <img
+                                                src={tip.src}
+                                                alt={tip.alt}
+                                                className="w-full h-auto rounded-b-none rounded-t-lg"
+                                            />
+                                            <div className="absolute bottom-0 flex items-center justify-center text-white bg-gradient-to-t from-black to-transparent text-center w-full p-4 py-6">
+                                                {tip.description}
+                                            </div>
+                                        </CardContent>
+                                    </Card>
+                                </div>
+                            </CarouselItem>
+                        ))}
+                    </CarouselContent>
+                    {showArrows && (
+                        <>
+                            <CarouselPrevious className="absolute left-0" />
+                            <CarouselNext className="absolute right-0" />
+                        </>
+                    )}
+                </Carousel>
+            )}
+            <div className="flex flex-col gap-4 text-center p-2">
+                <div className="text-center font-bold text-xl">
+                    Sign in to unlock your second brain
+                </div>
+            </div>
+            <div className="flex flex-col gap-8 pb-4 text-center align-middle items-center">
+                <GoogleSignIn onLoad={handleGoogleScriptLoad} />
+                {/* <div id="g_id_signin" /> */}
+                <Button
+                    variant="outline"
+                    className="w-[300px] p-8 flex gap-2 items-center justify-center rounded-lg font-bold"
+                    onClick={handleGoogleSignIn}
+                    disabled={isLoading || !data?.google}
```

Comment:
Shouldn't the Google sign-in button be disabled if google client_id and redirect_uri are unset as well?

```suggestion
                    disabled={isLoading || !data?.google || !data?.google.client_id || !data?.google.redirect_uri}
```